pkg/resource/stack/deployment.go

```diff
@@ -570,10 +570,37 @@ func DeserializePropertyValue(v interface{}, dec config.Decrypter,
 					}
 					urn := resource.URN(urnStr)
 
+					propertyValueID := func(v interface{}) (string, error) {
+						mapV, ok := v.(map[string]interface{})
+						if !ok {
+							return "", errors.New("malformed resource value: id must be a string")
+						}
+
+						V, ok := mapV["V"]
+						if !ok {
+							return "", errors.New("malformed resource value: id must be a string")
+						}
+
+						pv := resource.PropertyValue{V: V}
+						if pv.IsNull() {
+							return "", nil
+						}
```

Comment:
I'm not sure about this. If we return an empty string for `IsNull()`, the call to `resource.MakeCustomResourceReference(urn, resource.ID(id), packageVersion)` is going to set the id to be a computed value, instead of null.

https://github.com/pulumi/pulumi/blob/8f7dfbb5ae22062f9f10104cd4e155db5ef9172b/sdk/go/common/resource/properties.go#L243-L255

And Null is intended to represent component resources:

https://github.com/pulumi/pulumi/blob/8f7dfbb5ae22062f9f10104cd4e155db5ef9172b/sdk/go/common/resource/properties.go#L99-L113

It's not clear to me yet why you were seeing null for the ID of custom resources in the example state file you showed in Slack.

Reply:
> It's not clear to me yet why you were seeing null for the ID of custom resources in the example state file you showed in Slack.

It looks like the child resources of the ComponentResource are incorrectly being detected as ComponentResources.

Here are the versions I was using to reproduce this:
```
  <ItemGroup>
    <PackageReference Include="Pulumi" Version="2.17.0" />
    <PackageReference Include="Pulumi.Kubernetes" Version="2.7.0" />
  </ItemGroup>
```
CLI version `v2.18.0`